### ISSUE TITLE
Map type not returning transformed data as expected in test

### ISSUE THREAD

**Issue by clifton-mcintosh** [entropealabs/states_language#6]
### Environment

* Elixir version (elixir -v):
```
Erlang/OTP 22 [erts-10.4.4] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [hipe]

Elixir 1.9.1 (compiled with Erlang/OTP 22)
```

* Operating system:
MacOS Mojave
Version 10.14.6

### Expected behavior
I would expect the transformations in a Map type to be returned in the result from the state. With regard to the [specific test for Map](https://github.com/CityBaseInc/states_language/blob/master/test/states_language_map_test.exs), I would expect the "map_result" element to include items whose "data" element has been incremented by 10.

### Actual behavior
The results from the Map operation do not appear to get set in the "map_result" element in the test. Instead, the original data values (not incremented by 10) are present in that element.

When running the test `test/states_language_map_test.exs`, the test is passing, but the debug logs hint that un-transformed values are returned from the mapping operation. See here for a truncated version of the debug statements that run during `mix test test/states_language_map_test.exs`: 

[truncated-log-for-map-test.txt](https://github.com/CityBaseInc/states_language/files/4334493/truncated-log-for-map-test.txt)

Note the debug statements for "Sending Parent Result" and the "map_result" value in the end.

I've tried altering the test to check the final state of the data. This patch includes that alteration. The tests for making sure that the "data" elements were incremented fails. Here is a patch with the alteration (with .txt) extension to allow uploading via GitHub:

[verify_map.patch.txt](https://github.com/CityBaseInc/states_language/files/4334497/verify_map.patch.txt)

Here is the output from the failing test:

```
  1) test client goes through all states and runs parallel state machines (StatesLanguageMapTest)
     test/states_language_map_test.exs:51
     Assertion with == failed
     code:  assert map_result == [%{data: 11, owner: "Chris"}, %{data: 12, ownder: "Alex"}, %{data: 13, owner: "Gaby"}, %{data: 14, owner: "Oliver"}, %{data: 15, owner: "Finn"}, %{data: 16, owner: "Xander"}]
     left:  [%{owner: "Chris", data: 1}, %{ownder: "Alex", data: 2}, %{owner: "Gaby", data: 3}, %{owner: "Oliver", data: 4}, %{owner: "Finn", data: 5}, %{owner: "Xander", data: 6}]
     right: [%{owner: "Chris", data: 11}, %{ownder: "Alex", data: 12}, %{owner: "Gaby", data: 13}, %{owner: "Oliver", data: 14}, %{owner: "Finn", data: 15}, %{owner: "Xander", data: 16}]
     stacktrace:
       test/states_language_map_test.exs:67: (test)
```

Is this expected? If not, is this an artifact of the way the test was created or the way the states were defined in the supporting JSON files?

**Comment by entone** [comment]
Sorry I missed this somehow. @clifton-mcintosh I'll look into this.

**Comment by clifton-mcintosh** [comment]
Closing as resolved by #8 

@entone I look forward to an updated version of the library.

**Comment by entone** [comment]
I still need the hex ownership transferred to my hex user "entone"
